doc/specs/#492 - Default Terminal/spec.md

```diff
@@ -0,0 +1,248 @@
+    - **Not considering:** WinRT. `conhost.exe` has no WinRT. Adding WinRT to it would significantly increase the complexity of compilation in the inbox and out of box code base. It would also significantly increase the compilation time, binary size, library link list, etc... unless we use just the ABI to access it. But I don't see an advantage to that over just using classic COM at that point. This is only one handoff method and a rather simplistic one at that. Every benefit WinRT provides is outweighed by the extra effort that would be required over just a classic COM server in this case.
+- After delegation is complete, the inbox console will have to clean up any threads, handles, and state related to the session. We do a fairly good job with this normally, but some portions of the `conhost.exe` codebase are reliant on the process exiting for final cleanup. There may be a bit of extra effort to do some explicit cleanup here.
+
+#### Updated console
+
+The updated replacement console will have the same console API server capabilities as the inbox console, but will be a later, updated, or customized-to-the-scenario version of the API server generally revolving around improving ConPTY support for a Terminal application.
+
+On receiving the handoff from the method signature listed above, the updated console will:
+- Establish its own set of IO threading, device communication infrastructure, and API messaging routines while storing the handles given
+- Re-parse the command line arguments, if necessary, and store them for guiding the remainder of launch
+- Dispatch the attach message as if it were received normally
+- Continue execution from there
+
+There will then either be a registration for a Terminal UX to take over the session by using ConPTY, or the updated console will choose to launch its potentially updated version of the `conhost` UX.
+
+For registration, we repeat the dance above with another key:
+- A registry key in `HKCU\Console` (format `REG_SZ`, name `DelegationTerminal`) would specify the path to the replacement console that would be used to service the remainder of the connection process.
+    - Alternatively or additionally, this same `REG_SZ` could list a COM server ID that could be looked up in the classes root and invoked
+        - Packaged applications and classic applications can easily register a COM server
+        - **To be confirmed:** WinRT libraries should be able to be easily registered as the COM server as well (given WinRT is COM underneath)
+
+The delegation repeats the same dance as above as well:
+- A contract (interface) is established between the updated console and the terminal
+    - **METHOD SIGNATURE NOT DETERMINED**

```

Comment:
Right now I have this as a command line argument and no formal structure... that's not great but not unusual for console/terminal initialization stuff.

Reply:
It turned into a COM function.